### ISSUE TITLE
HtsFile not showing genome assembly

### ISSUE THREAD

**Issue by pnrobinson** [phenopackets/phenopacket-schema#38]
If I create a phenopacket that includes this,
```
 HtsFile htsFile = HtsFile.newBuilder()
                .setHtsFormat(HtsFile.HtsFormat.VCF)
                .setGenomeAssembly(org.phenopackets.schema.v1.core.GenomeAssembly.GRCH_37)
                // in this case the proband identifier is different to the sample
                // identifier used in the VCF file
                .putIndividualToSampleIdentifiers(proband.getId(), "manuel")
                .setFile(File.newBuilder().setPath("/home/robinp/data/exomiser-cli-9.0.1/examples/Pfeiffer.vcf").build())
                .build();
```
then the output is missing the part about "GRCH_37"
``` "htsFiles": [{
    "htsFormat": "VCF",
    "individualToSampleIdentifiers": {
      "manuel": "manuel"
    },
    "file": {
      "path": "/home/robinp/data/exomiser-cli-9.0.1/examples/Pfeiffer.vcf"
    }
  }],
```
Is my code wrong or what is the way of indicating the genome build?

**Comment by julesjacobsen** [comment]
I think the default is not to include the data if the value is the default. If you set it the genome assembly the GRCH_38 you ought to see this in the json output.

**Comment by pnrobinson** [comment]
OK! I would suggest that we not have a default for the genome build because there is no default in the community, and even if we say that h37 is the default now, this will change next year, and thus the same PhenoPacket file will have a different meaning depending on when we read it. And asking client software to keep track of when defaults for genome builds change is not robust.

**Comment by mattions** [comment]
We see a lot of projects already using GRCH_38, so, unless there is no a clear need, I'm also with @pnrobinson and avoiding to set any default.

**Comment by julesjacobsen** [comment]
Can't not have a default. It's a protobuf requirement for an enum and it's simply a default value for encoding/decoding. Client code should always set the genome build - if you use the getter the GRCH_37 value is returned.

**Comment by pnrobinson** [comment]
Can I suggest that we make the default "UNKNOWN" -- this will presumably force the output file to have either 37 or 38 (or 39....) if somebody does set the genome. I think that otherwise there is a really big potential for confusion.

**Comment by julesjacobsen** [comment]
OK seems sensible - this will be a breaking and completely incompatible change by the way - GRCH_37 messages from an older client will be mis-interpreted as UNKNOWN. Unless there is an option somewhere to force json output to write defaults, which might mitigate some confusion.

**Comment by pnrobinson** [comment]
Given that some confusion seems inevitable, it seems best to fail with some UNKNOWN GENOME BUILD error rather than to silently use the wrong genome once the default changes...

**Comment by julesjacobsen** [comment]
Peter, I don't think this is how it works. There is an option to enable printing of default fields, so your output will become:

```json
  "htsFormat": "VCF",
  "genomeAssembly": "GRCH_37",
  "individualToSampleIdentifiers": {
      "manuel": "manuel"
    },
    "file": {
      "path": "/home/robinp/data/exomiser-cli-9.0.1/examples/Pfeiffer.vcf"
    }
  }],
```
I can add that as a fix for this to avoid confusion when converting to JSON. 
You can enable this by replacing ```PhenoPacketFormat.toJson(phenoPacket)``` with the underlying implementation:

```java
JsonFormat.printer().includingDefaultValueFields().print(phenoPacket);
```

**Comment by cmungall** [comment]
Pointless whinge: this is one area where a plain json-schema or similar approach seems better than protobuf. The behavior with enums seems a bit bloody terrifying:

http://androiddevblog.com/protocol-buffers-pitfall-adding-enum-values/

Personally, whenever I see an enum I want to replace it with an ontology term, even if that seems overkill, and that would avoid some of the issues here. It would also mean we don't have to keep changing these schema every time there is a new build. Granted not that often.

But if we stick with enums, it seems we should future-proof them all, and make a one time breaking change to include an UNKNOWN for all enums in the schema.

**Comment by cmungall** [comment]
Hmm I guess this is sort of fixed in proto3, but the recommendation is to have a default zero valued enum

https://developers.google.com/protocol-buffers/docs/proto3#enum

**Comment by julesjacobsen** [comment]
Yeah, it's not ideal. Avro doesn't quite suffer from this, but has other issues. json-schema wouldn't suffer from this either, but it doesn't provide as many language bindings, so pick your poison...

Are we all agreed on adding a breaking UNKNOWN to all enums with the default value of 0 being added and always printing unknown fields?

**Comment by cmungall** [comment]
>
> Are we all agreed on adding a breaking UNKNOWN to all enums with the
> default value of 0 being added and always printing unknown fields?
>
Seems safest

**Comment by julesjacobsen** [comment]
Fixed in 2e8447988d21d2f9d04ef8f60579c798e8224bde
